lib/dockerregistry/types.go

```diff
@@ -60,6 +60,12 @@ type SyncContext struct {
 	ParentDigest      ParentDigest
 }
 
+// PreCheck represents a check function to run against a pull request that
+// modifies the promoter manifests before oking promotion of the changes.
```

Comment:
```
// PreCheck is the interface that wraps the basic Run method.
//
// <... explain in more detail here what Run() does>
```

Take a look at the official Golang library docs to get a sense of the commenting style, for example https://golang.org/src/io/io.go#L115